shop4All-backend/src/main/java/com/example/shop4All_backend/services/CategoryService.java

```diff
@@ -0,0 +1,52 @@
+package com.example.shop4All_backend.services;
+
+import com.example.shop4All_backend.entities.Category;
+import com.example.shop4All_backend.entities.Product;
+import com.example.shop4All_backend.exceptions.UserException;
+import com.example.shop4All_backend.repositories.CategoryRepo;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+import java.util.HashSet;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+@Service
+public class CategoryService {
+
+    @Autowired
+    private CategoryRepo categoryRepo;
```

Comment:
I think it is better to make this variable final and then use @RequiredArgsConstructor annotation above the CategoryService class because you don't want to reassign the categoryRepo variable.